presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -52,29 +64,35 @@
 /**
  * NativeExecutionOperator is responsible for launching the external native process and managing the communication
  * between Java process and native process (by using the {@Link NativeExecutionTask}). The NativeExecutionOperator will send  necessary meta information
- * (e.g, plan fragment, session properties etc) will be sent to native process and collect the execution results (data, metrics etc) back and propagate out as
+ * (e.g, plan fragment, session properties etc.) will be sent to native process and collect the execution results (data, metrics etc) back and propagate out as
  * the operator output through the operator's getOutput method.
+ * The lifecycle of the NativeExecutionOperator is:
+ * 1. Launch the native engine external process when initializing the operator.
+ * 2. Serialize and pass the planFragment, tableWriteInfo and taskSource to the external process through {@link NativeExecutionTask} APIs.
+ * 3. Call {@link NativeExecutionTask}'s pollResult() to retrieve {@link SerializedPage} back from external process.
+ * 4. Deserialize {@link SerializedPage} to {@link Page} and return it back to driver from the getOutput method.
+ * 5. The close() will be called by the driver when {@link NativeExecutionTask} completes and pollResult() returns an empty result.
+ * 6. Shut down the external process upon calling of close() method
  * <p>
- * TODO: The lifecycle of NativeExecutionOperator is (will be added in next PR):
- *  1. Launch the native engine external process when initializing the operator.
- *  2. Serialize and pass the planFragment, tableWriteInfo and taskSource to the external process through {@link NativeExecutionTask} APIs.
- *  3. Call {@link NativeExecutionTask}'s pollResult() to retrieve {@link SerializedPage} back from external process.
- *  4. Deserialize {@link SerializedPage} to {@link Page} and return it back to driver from the getOutput method.
- *  5. The close() will be called by the driver when {@link NativeExecutionTask} completes and pollResult() returns an empty result.
- *  6. Shut down the external process upon calling of close() method
  */
 public class NativeExecutionOperator
         implements SourceOperator
 {
+    private static final Logger log = Logger.get(NativeExecutionOperator.class);
+    private static final String NATIVE_EXECUTION_SERVER_URI = "http://127.0.0.1";
```

Comment:
Does IPV4 vs IPV6 matter here ?

Reply:
I think presto_cpp is listening on IPV4 address - @tanjialiang may have more context.

Reply:
Here it is in the config.properties file
discovery.uri=http://127.0.0.1:58215

Reply:
So I believe it listens on IPV4